bridge_adaptivity/module/static/module/js/module.js

```diff
@@ -360,9 +360,12 @@
         const modalSuccessListener = (url) => {
             return result => {
                 let modal = $('#modal-wrapper');
-                if (result.status !== undefined && result.status === 'ok' && result.collection_slug !== undefined) {
-                    var new_url = new URL(window.location.href);
+                var new_url = new URL(window.location.href);
+                 // FixMe: This check  is needed to remove in another place
+                if (result.collection_slug !== undefined) {
                     new_url.searchParams.append('collection_slug', result.collection_slug);
+                }
+                if (result.status !== undefined && result.status === 'ok') {
```

Comment:
Why do we need two statements it seems that second covers both cases.

Reply:
The first statement is used when the collection id changed. And it's used by only one flow: When we use CollectionUpdate view
The second state is used by all flows to create or update something. When a form saved correctly, we could reload the page.

Reply:
I meant `result.status !== undefine` is totally the same as we check `result.status === 'ok'`. So why do we need the first check?

Reply:
You are right! We don't need the first check. I fixed this.